test/integration_test/migration_test.go

```diff
@@ -568,3 +575,87 @@ func migrationScheduleTest(
 	_, err = task.DoRetryWithTimeout(f, defaultWaitTimeout, defaultWaitInterval)
 	require.NoError(t, err, "migrations for schedules have not been deleted")
 }
+
+func migrationScaleTest(t *testing.T) {
+	ctxs := make([]*scheduler.Context, migrationScaleCount)
+	var allMigrations []*v1alpha1.Migration
+	var err error
+	// Reset config in case of error
+	defer func() {
+		err = setRemoteConfig("")
+		require.NoError(t, err, "Error resetting remote config")
+	}()
+	migrationKey := "mysql-migration"
+	migrationAppKey := "mysql-1-pvc"
+	includeResourcesFlag := true
+
+	for i := 1; i <= migrationScaleCount; i++ {
+		currCtxs, err := schedulerDriver.Schedule(migrationKey+"-"+strconv.Itoa(i),
+			scheduler.ScheduleOptions{AppKeys: []string{migrationAppKey}})
+		require.NoError(t, err, "Error scheduling task")
+		require.Equal(t, 1, len(currCtxs), "Only one task should have started")
+
+		err = schedulerDriver.WaitForRunning(currCtxs[0], defaultWaitTimeout, defaultWaitInterval)
+		require.NoError(t, err, "Error waiting for app to get to running state")
+
+		// create, apply and validate cluster pair specs
+		err = scheduleClusterPair(currCtxs[0], false)
+		require.NoError(t, err, "Error scheduling cluster pair")
+
+		ctxs = append(ctxs, currCtxs...)
+
+		currMigNamespace := migrationAppKey + "-" + migrationKey + "-" + strconv.Itoa(i)
+		currMig, err := createMigration(t, migrationKey, currMigNamespace, "remoteclusterpair", currMigNamespace, &includeResourcesFlag, true)
+		logrus.Infof("Successfully created migration %s:%s", currMig.Name, currMig.Namespace)
+		allMigrations = append(allMigrations, currMig)
+
+	}
+	err = WaitForMigration(allMigrations)
```

Comment:
You should be able to loop on all the migrations and call `validateAndDestroyMigration()` which will also take care of deleting the migrations after they succeed.

Reply:
Validation also expects same migration to be present, since we have diff migrations we cannot use that here.